backend/degree/views.py

```diff
@@ -189,3 +193,69 @@ def create(self, request, *args, **kwargs):
             return self.partial_update(request, *args, **kwargs)
         except Http404:
             return super().create(request, *args, **kwargs)
+    
+
+class DegreeProfileViewset(viewsets.ModelViewSet):
+    permission_classes = [IsAuthenticated]
+    serializer_class = DegreeProfileSerializer
+
+    def get_queryset(self):
+        return DegreeProfile.objects.filter(user_profile__user=self.request.user)
+
+    def get_serializer_class(self):
+        if self.request.method == "PATCH":
+            return DegreeProfilePatchSerializer
+        return super().get_serializer_class()
+
+    def perform_create(self, serializer):
+        user_profile, _ = UserProfile.objects.get_or_create(user=self.request.user)
+        return serializer.save(user_profile=user_profile.id)
+
+    def update(self, request, *args, **kwargs):
```

Comment:
if we have a ModelViewset, the nice part is we don't have to define most of these things. See e.g., the SectionList view. This is because django rest framework ModelViewset already has defaults for these things.